src/frontend/js/utils/OrderHelper/index.ts

```diff
@@ -14,10 +20,10 @@ export class OrderHelper {
     return orders.find(filter);
   }
 
-  static orderNeedsSignature(order: Order, product?: Product) {
+  static orderNeedsSignature(order: Order, contract_definition?: ContractDefinition) {
```

Comment:
The fact is that we need to know  if a order needs to be signed without having a full product return by the api.
And it will be a shame to fetch this product just for this.

I think i may totaly remove `contractDefinition` argument as for now we only check if it's undefined or not, that can be done in the parent, before calling `orderNeedsSignature`.

A other solution that i feel more complex is to create a `ProductSignature` interface with only `contract_definition` and *may be other stuff* in the future. 
Then make `Product` extends this new `ProductSignature`
Then use it here. instead of `Product` or `ContractDefinition`.

We don't know if and how signature would change. I think we should make it simple without anticipate random stuff.
so either: 
* removing this argument and add a `!!contract_definition` before calling `orderNeedsSignature`
* keep this new `contract_definition` argument instead of product to force developer to check it when using `orderNeedsSignature`

Reply:
Ok, if the API does not return a full product it seems we have no choice anyway